seqr/migrations/0063_dynamicanalysisgroup.py

```diff
@@ -0,0 +1,33 @@
+# Generated by Django 3.2.23 on 2024-04-08 20:54
+
+from django.conf import settings
+from django.db import migrations, models
+import django.db.models.deletion
+import django.utils.timezone
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
+        ('seqr', '0062_individual_solve_status'),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name='DynamicAnalysisGroup',
+            fields=[
+                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('guid', models.CharField(db_index=True, max_length=30, unique=True)),
+                ('created_date', models.DateTimeField(db_index=True, default=django.utils.timezone.now)),
+                ('last_modified_date', models.DateTimeField(blank=True, db_index=True, null=True)),
+                ('name', models.TextField()),
+                ('criteria', models.JSONField()),
+                ('created_by', models.ForeignKey(blank=True, null=True, on_delete=django.db.models.deletion.SET_NULL, related_name='+', to=settings.AUTH_USER_MODEL)),
+                ('project', models.ForeignKey(blank=True, null=True, on_delete=django.db.models.deletion.CASCADE, to='seqr.project')),
+            ],
+            options={
+                'unique_together': {('project', 'name')},
```

Comment:
how does this constraint behave if project is null?

Reply:
It means you can only have a name associated with one list with no project. So fi there is already a list of that name associated with a different project its fine, but if it is associated with no project it would not let you make a duplicate